projects/geode/index.js

```diff
@@ -0,0 +1,98 @@
+const sdk = require("@defillama/sdk");
+const BigNumber = require("bignumber.js");
+
+const PORTAL = require("./abis/avax/Portal.json");
+const gAVAX = require("./abis/avax/gAVAX.json");
+const DWP = require("./abis/avax/DWP.json");
+
+const transformAddress = (addr) => `avax:${addr}`;
+
+async function avax(timestamp, ethBlock, chainBlocks) {
+  const chain = "avax";
+  const block = chainBlocks[chain];
+
+  const planetType = 5;
+
+  const planetIds = (
+    await sdk.api.abi.call({
+      block: block,
+      target: PORTAL.address,
+      params: planetType,
+      abi: PORTAL.abi.find((abi) => abi.name === "getIdsByType"),
+      chain,
+    })
+  ).output;
+
+  // Find hosted derivative total Supplies, multiply with pricePerShare
+  const supplies = (
+    await sdk.api.abi.multiCall({
+      block,
+      calls: planetIds.map((id) => ({
+        params: id,
+      })),
+      target: gAVAX.address,
+      abi: gAVAX.abi.find((abi) => abi.name === "totalSupply"),
+      chain,
+    })
+  ).output;
+
+  const prices = (
+    await sdk.api.abi.multiCall({
+      block,
+      calls: planetIds.map((id) => ({
+        params: id,
+      })),
+      target: gAVAX.address,
+      abi: gAVAX.abi.find((abi) => abi.name === "pricePerShare"),
+      chain,
+    })
+  ).output;
+
+  // Find DWP addresses and count only Avax(index:0) balances, excluding gAVAX(index:1) balances
+  const dwpAddresses = (
+    await sdk.api.abi.multiCall({
+      block,
+      calls: planetIds.map((id) => ({
+        params: id,
+      })),
+      target: PORTAL.address,
+      abi: PORTAL.abi.find((abi) => abi.name === "planetWithdrawalPool"),
+      chain,
+    })
+  ).output;
+
+  const dwpOwnedIdle = (
```

Comment:
what are dwps?

Reply:
Dynamic Withdrawal Pools is a stableswap pool with dynamic peg. Allowing low-slippage instant withdrawals for Staking Derivatives such as yyAVAX. More: https://docs.geode.fi/fundamentals/understanding-the-dynamic-withdrawals/deep-dive-withdrawal-pools

Reply:
These pools include both idle and staked asset. Calculating only includes the idle asset (Avax) as derivative is already counted with Total Supply calculations above.